### PR TITLE
Add an API for querying the information needed for an analysis from a…

### PR DESCRIPTION
**Pull request by dpryan79**
… flow cell

### CODE REVIEW

**Comment by dpryan79** [comment]
@anatskiy For some reason I always get `{"detail":"Authentication credentials were not provided."}` and error code 403 when I try to request this via python (`res = requests.get("http://localhost/api/analysis_list/analysis_list/", auth=(<user>, <pass>), params=d)`), but it works perfectly fine if I query it with a browser. Any clue why?

**Comment by anatskiy** [comment]
@dpryan79 could you provide the class/method which is responsible for the API endpoint `api/analysis_list/analysis_list/`?

File: wui/api.py [hunk 1]
```diff
@@ -61,3 +61,4 @@
 router.register(r'sequencing_costs', SequencingCostsViewSet, base_name='sequencing-costs')
 
 router.register(r'run_statistics', RunStatisticsViewSet, base_name='run-statistics')
+router.register(r'analysis_list', FlowcellViewSet, base_name='analysis_list')
```
Comment (anatskiy) [inline wui/api.py]: you shouldn't include it here as it's already included on the line 55

**Comment by anatskiy** [comment]
@dpryan79 try to change the url to `api/flowcells/analysis_list/`

File: wui/api.py (hunk 1 above)
Comment (dpryan79) [reply wui/api.py]: Changed

**Comment by dpryan79** [comment]
Doesn't have any effect. The modified URL works in a browser but not via python. Very strange.

**Comment by anatskiy** [comment]
Yeah, that's interesting. I'll take a look right now.

**Comment by anatskiy** [comment]
@dpryan79 oh, I found out what causes the issue: line 95 `authentication_classes = [CsrfExemptSessionAuthentication]`

If one removes it, the files (Benchtop Protocol and Sample Sheet) couldn't be downloaded again. I'm aware of this problem but don't have a solution yet. I need more time to find a workaround.

File: flowcell/views.py [hunk 2]
```diff
@@ -356,6 +356,15 @@ def create_row(lane, record):
     @list_route(methods=['get'])
     def analysis_list(self, request):
         flowcell_id = request.query_params.get('flowcell_id', '')
+
+        # Don't throw an error if the flow cell ID is missing
```
Comment (anatskiy) [inline flowcell/views.py]: this all can be replaced with just one line :)
```
from django.shortcuts import get_object_or_404

...

flowcell = get_object_or_404(Flowcell, flowcell_id=flowcell_id)
```

Comment (dpryan79) [reply flowcell/views.py]: Ah, figures :)

**Comment by dpryan79** [comment]
OK, thanks for finding the cause at least!

**Comment by anatskiy** [comment]
@dpryan79 if this feature is urgent, then creating a separate viewset (and including it to `wui/api.py`) can be a temporary solution. I could do it in the evening.

**Comment by dpryan79** [comment]
I wouldn't call it urgent. I can also look at other viewsets, since I could presumably add this to a project-oriented viewset. At some point we just need an API to allow querying the organisms and library type used by either flow cell or project, so we can trigger analysis pipelines accordingly.

**Comment by dpryan79** [comment]
@anatskiy I followed your suggestion of just making a different view set. That seems to have worked easily enough.

**Comment by dpryan79** [comment]
@anatskiy I've merged `develop` into this so I'll make this branch the live one on parkour.ie-freiburg.mpg.de so I can get the analysis stuff set up. Let me know if you need a different branch to be the live one so I can revert that.

**Comment by dpryan79** [comment]
@anatskiy This is now working properly on the live version. Can you have a look and merge if it seems OK?

**Comment by anatskiy** [comment]
@dpryan79 I think I have a solution to the problem we had before when we tried to merge it with an already existing Viewset. I'll test this out. If I succeed, then I'll push a few changes and will merge it.

**Pull request merged**